app/models/distribution.rb

```diff
@@ -54,9 +54,15 @@ class Distribution < ApplicationRecord
   scope :recent, ->(count = 3) { order(issued_at: :desc).limit(count) }
   scope :future, -> { where("issued_at >= :tomorrow", tomorrow: Time.zone.tomorrow) }
   scope :during, ->(range) { where(distributions: { issued_at: range }) }
-  scope :for_csv_export, ->(organization) {
+  scope :for_csv_export, ->(organization, filters = {}, date_range = nil) {
     where(organization: organization)
       .includes(:partner, :storage_location, :line_items)
+      .apply_filters(filters, date_range)
+  }
+  scope :apply_filters, ->(filters, date_range) {
+    includes(:partner, :storage_location, :line_items, :items)
+      .order(issued_at: :desc)
+      .class_filter(filters.merge(during: date_range))
```

Comment:
Could you elaborate on this? I'm not sure I understand the approach here?

If it's necessary to solve it in this way, could you document the reasoning behind your process, so that it will be clearer on how to maintain it?

Reply:
@armahillo what part do you mean? the merge? That could be a direct scope call, actually.

Reply:
I took the approach of merging it into the filters passed to `class_filter` because that would already include the behavior of ignoring the filter when a date range is not given